src/main/java/org/codehaus/mojo/jaxws/WsImportMojo.java

```diff
@@ -392,7 +392,8 @@ else if ( settings != null )
             String nonProxyHostsString = getActiveNonProxyHosts( settings );
             if ( nonProxyHostsString != null )
             {
-                addVmArg( "-Dhttp.nonProxyHosts=" + nonProxyHostsString.replace( "|", "^|" ) );
+                final String argument = String.format( "-Dhttp.nonProxyHosts=%s", escapeArgumentPart( nonProxyHostsString ) );
+                addVmArg( argument );
             }
```

Comment:
Instead of escaping a single special character here, I've escaped the complete nonProxyHosts setting. This is due to several special characters that are allowed in the syntax of the setting like:

| .. * [ ], etc.

Reply:
Okay point taken here - because I don't like "String.format" I will replace it with regular String concatenation.

Reply:
Using String.format or not is no big deal for me. What works, works and I don't see much difference in code quality here. So go for it.